app/(candidate)/dashboard/text-messaging/p2p-setup/[step]/components/ReviewStep.js

```diff
@@ -1,33 +1,56 @@
 'use client'
 import Button from '@shared/buttons/Button'
-import { useCampaign } from '@shared/hooks/useCampaign'
 import Body1 from '@shared/typography/Body1'
 import Body2 from '@shared/typography/Body2'
 import H2 from '@shared/typography/H2'
 import Overline from '@shared/typography/Overline'
 import Paper from '@shared/utils/Paper'
 import { AlertBanner } from 'app/(candidate)/dashboard/components/AlertBanner'
-import { updateCampaign } from 'app/(candidate)/onboarding/shared/ajaxActions'
+import { clientFetch } from 'gpApi/clientFetch'
+import { apiRoutes } from 'gpApi/routes'
 import Link from 'next/link'
 import { useRouter } from 'next/navigation'
+import { useComplianceForm } from './ComplianceFormContext'
+import { useSnackbar } from 'helpers/useSnackbar'
+
+/**
+ * Submits compliance information for text messaging campaign
+ * @param {Object} body - The compliance information
+ * @param {string} body.ein - Employer Identification Number
+ * @param {string} body.name - Org name associated with EIN
+ * @param {string} body.address - Address associated with EIN
+ * @param {string} body.website - 10DLC compliant website
+ * @param {string} body.email - 10DLC compliant email
+ */
+function submitCompliance(body) {
+  return clientFetch(apiRoutes.textMessaging.submitCompliance, {
+    ...body,
+  })
+}
 
 export default function ReviewStep() {
-  const [campaign] = useCampaign()
   const router = useRouter()
+  const [complianceForm] = useComplianceForm()
```

Comment:
If the user reloads the page here, they lose all this data, no? 🤔 

Unfortunately, in order to avoid a broken UX on page reload, I think we need to persist this still 😕